libraries/ms-common/src/stm32f0xx/spi.c

```diff
@@ -24,7 +24,7 @@ StatusCode spi_init(SPIPort spi, const SPISettings *settings) {
   RCC_ClocksTypeDef clocks;
   RCC_GetClocksFreq(&clocks);
 
-  size_t index = (size_t)__builtin_ffsl((int32_t)(clocks.PCLK_Frequency / settings->baudrate));
+  size_t index = 32 - (size_t)__builtin_clz(clocks.PCLK_Frequency / settings->baudrate);
```

Comment:
So to clarify what's going on here:

Since the prescaler must be one of these values

```c
#define SPI_BaudRatePrescaler_2         ((uint16_t)0x0000) // 00 = 0 << 3
#define SPI_BaudRatePrescaler_4         ((uint16_t)0x0008) // 08 = 1 << 3
#define SPI_BaudRatePrescaler_8         ((uint16_t)0x0010) // 16 = 2 << 3
#define SPI_BaudRatePrescaler_16        ((uint16_t)0x0018) // 24 = 3 << 3
#define SPI_BaudRatePrescaler_32        ((uint16_t)0x0020) // 32 = 4 << 3
#define SPI_BaudRatePrescaler_64        ((uint16_t)0x0028) // 40 = 5 << 3
#define SPI_BaudRatePrescaler_128       ((uint16_t)0x0030) // 48 = 6 << 3
#define SPI_BaudRatePrescaler_256       ((uint16_t)0x0038) // 56 = 7 << 3
```

we divide the Peripheral Clock by the baudrate, to get the prescaler value. And then we want the index of the first set bit (from the right), which we use to find largest power of 2 that fits in (to find the closest prescaler that matches).

If this is the case, I think the check for `index` should be `index < 2` and not `index <= 2`, since `SPI_BaudRatePrescaler_2` is `#define`'d to `0x00`